File: feature/harry_potter/src/main/java/com/example/lego/feature/harrypotter/HarryPotterViewModel.kt
package com.example.lego.feature.harrypotter

import androidx.lifecycle.ViewModel
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject

import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow

data class HPCharacter(
    val id: String,
    val name: String,
    val house: String,
    val actor: String
)

@HiltViewModel
class HarryPotterViewModel @Inject constructor() : ViewModel() {

    private val _characters = MutableStateFlow<List<HPCharacter>>(emptyList())
    val characters: StateFlow<List<HPCharacter>> = _characters

    init {
        // Load mock data
        _characters.value = listOf(
            HPCharacter("1", "Harry Potter", "Gryffindor", "Daniel Radcliffe"),
            HPCharacter("2", "Hermione Granger", "Gryffindor", "Emma Watson"),
            HPCharacter("3", "Ron Weasley", "Gryffindor", "Rupert Grint"),
            HPCharacter("4", "Draco Malfoy", "Slytherin", "Tom Felton"),
            HPCharacter("5", "Severus Snape", "Slytherin", "Alan Rickman")
        )
    }

    fun getCharacterById(id: String): HPCharacter? {
        return _characters.value.find { it.id == id }
    }
}

File: Sources/Features/HarryPotter/HarryPotterViewModel.swift
import SwiftUI

struct HPCharacter: Identifiable {
    let id: String
    let name: String
    let house: String
    let actor: String
}

class HarryPotterViewModel: ObservableObject {
    @Published var characters: [HPCharacter] = []
    
    init() {
        characters = [
            HPCharacter(id: "1", name: "Harry Potter", house: "Gryffindor", actor: "Daniel Radcliffe"),
            HPCharacter(id: "2", name: "Hermione Granger", house: "Gryffindor", actor: "Emma Watson"),
            HPCharacter(id: "3", name: "Ron Weasley", house: "Gryffindor", actor: "Rupert Grint"),
            HPCharacter(id: "4", name: "Draco Malfoy", house: "Slytherin", actor: "Tom Felton"),
            HPCharacter(id: "5", name: "Severus Snape", house: "Slytherin", actor: "Alan Rickman")
        ]
    }
    
    func getCharacterById(id: String) -> HPCharacter? {
        characters.first { $0.id == id }
    }
}
